### ISSUE TITLE
ConfigMap update for explicit types

### ISSUE THREAD

**Issue by Verticaleap** [findelabs/proxima#36]
We should not assume types, instead we should set endpoint types explicitly:
```yaml
routes:
  endpoint:
    proxy:
      url:
  endpoint:
    vault:
      secret:
  endpoint: 
    http_config:
      url:
```

**Comment by Verticaleap** [comment]
Once this is complete, we should remove the url failover variant, as we can just specify a endpoint with failover type instead.

**Comment by Verticaleap** [comment]
Released in 0.8.0
